### ISSUE TITLE
Issues with positioning point lights

### ISSUE THREAD

**Issue by mindofmatthew** [processing/p5.js#1622]
Right now, point light positioning doesn't use the same coordinate space as everything else. In the Processing example below, the light is positioned between the two spheres. In the p5 version of the same example, the light is positioned far to the left of the spheres (I'm pretty sure that its effective x-position is -40000, rather than -40).

Processing example:
![lights in Processing](https://cloud.githubusercontent.com/assets/911429/19174689/c7a1b974-8bfe-11e6-9716-ff6745830f23.png)

``` processing
size(200, 100, P3D);

noStroke();
fill(255);
background(0);

ortho();

translate(width / 2, height / 2, 0);

pointLight(255, 255, 255, -40, 0, 0);

sphere(20);

pushMatrix();
translate(-80, 0, 0);
sphere(20);
popMatrix();
```

p5 Example:
![lights in p5.js](https://cloud.githubusercontent.com/assets/911429/19174910/5bd1a6ee-8c00-11e6-8f74-8f087885b02d.png)

``` javascript
function setup() {
  createCanvas(200, 100, WEBGL);
}

function draw() {
  background(0);
  ambientMaterial(255, 255, 255);

  pointLight(255, 255, 255, -40, 0, 0);

  sphere(20);

  push();
  translate(-80, 0, 0);
  sphere(20);
  pop();
}
```

**Comment by kjhollen** [comment]
I think this one will be handled with #2139!

**Comment by kjhollen** [comment]
Just an update to say that this wasn't fixed with the coordinate system like I'd hoped. I'll take a closer look at the lighting shader to try and see what's up.

**Comment by Spongman** [comment]
here's another example that demonstrates this problem: https://codepen.io/Spongman/pen/xpEpqd?editors=0010

the large sphere is lit by a point light at the location of the small sphere. the small sphere is on the x axis. only the right half of the large sphere should be lit. Also note that the lower sphere is lit identically to the other one.

![image](https://user-images.githubusercontent.com/1088194/34290409-07fff252-e6ac-11e7-95d1-760d455a17e8.png)

I believe the problem is caused by the fact that the uPointLightLocation positions are in model space and the rest of the calculation is done in camera. Unfortunately this cannot simply be fixed by multiplying the uPointLightLocation by the MV matrix as this will cause the lights to also be affected by the translate/rotate calls.

i believe the final fix involve separating the view & model matrices.

**Comment by Spongman** [comment]
ok, this is now fixed now that #2428 has been merged.

@mindofmatthew's original p5.js example needs a call to `ortho()` in the `setup()` function in order to make it look like the java version.

**Comment by kjhollen** [comment]
closing this as fixed per comment from @Spongman above. thank you!
